python/ray/data/tests/test_dataset_image.py

```diff
@@ -1,193 +1,142 @@
-    ds = ray.data.read_datasource(ImageFolderDatasource(), root=root)
-    arrays = ds.to_pandas()["image"]
-    shapes = sorted(array.shape for array in arrays)
-    assert shapes == [(16, 16, 3), (32, 32, 3), (64, 64, 3)]
-
-
-@pytest.mark.parametrize(
-    "mode, expected_shape", [("L", (32, 32)), ("RGB", (32, 32, 3))]
-)
-def test_image_folder_datasource_mode_parameter(
-    mode,
-    expected_shape,
-    ray_start_regular_shared,
-    enable_automatic_tensor_extension_cast,
-):
-    """Test `ImageFolderDatasource` works with images from different colorspaces.
-
-    The folder "different-modes" contains two cat images and one dog image. Their modes
-    are "CMYK", "L", and "RGB". All images are 32x32.
-    """
-    root = "example://image-folders/different-modes"
-    ds = ray.data.read_datasource(ImageFolderDatasource(), root=root, mode=mode)
-
-    tensors = ds.to_pandas()["image"]
-    assert all([tensor.shape == expected_shape for tensor in tensors])
-
-
-@pytest.mark.parametrize("size", [(-32, 32), (32, -32), (-32, -32)])
-def test_image_folder_datasource_value_error(ray_start_regular_shared, size):
-    root = "example://image-folders/simple"
-    with pytest.raises(ValueError):
-        ray.data.read_datasource(ImageFolderDatasource(), root=root, size=size)
-
-
-def test_image_folder_datasource_e2e(shutdown_only):
-    from ray.train.torch import TorchCheckpoint, TorchPredictor
-    from ray.train.batch_predictor import BatchPredictor
-
-    from torchvision import transforms
-    from torchvision.models import resnet18
-
-    root = "example://image-folders/simple"
-    dataset = ray.data.read_datasource(
-        ImageFolderDatasource(), root=root, size=(32, 32)
+class TestReadImages:
+    def test_basic(self, ray_start_regular_shared):
+        """Test basic `read_images` functionality.
+        The folder "simple" contains three 32x32 RGB images.
+        """
+        ds = ray.data.read_images("example://image-datasets/simple")
+        assert ds.schema().names == [TENSOR_COLUMN_NAME]
+        column_type = ds.schema().types[0]
+        assert isinstance(column_type, ArrowTensorType)
+        assert all(array.shape == (32, 32, 3) for array in ds.take())
+
+    def test_filtering(self, ray_start_regular_shared):
+        """Test `read_images` correctly filters non-image files.
+        The folder "different-extensions" contains three 32x32 RGB images and two
+        non-images.
+        """
+        ds = ray.data.read_images("example://image-datasets/different-extensions")
+        assert ds.count() == 3
+
+    def test_size(self, ray_start_regular_shared):
+        """Test `read_images` size parameter works with differently-sized images.
+        The folder "different-sizes" contains three RGB images. Each image has a
+        different size, with the size described in file names (e.g., 32x32.png).
+        """
+        ds = ray.data.read_images(
+            "example://image-datasets/different-sizes", size=(32, 32)
+        )
```

Comment:
Nit: You should also test that reading images of different sizes without specifying a `size` currently raises an error.

Reply:
Now that #29071 is merged, I've added a test that `read_images` on different sizes w/o `size` works as expected